Mathlib/Algebra/Polynomial/AlgebraMap.lean

```diff
@@ -678,10 +679,16 @@ theorem nmem_nonZeroDivisors_iff {P : R[X]} : P ∉ R[X]⁰ ↔ ∃ a : R, a ≠
   contrapose! ha
   exact h a ha
 
-open nonZeroDivisors in
 protected lemma mem_nonZeroDivisors_iff {P : R[X]} : P ∈ R[X]⁰ ↔ ∀ a : R, a • P = 0 → a = 0 := by
   simpa [not_imp_not] using (nmem_nonZeroDivisors_iff (P := P)).not
 
+lemma mem_nonzeroDivisors_of_coeff_mem {p : R[X]} (n : ℕ) (hp : p.coeff n ∈ R⁰) :
```

Comment:
Can you add the reverse direction? `P` is not a zero divisors iff there exist a coeff that is not a zero divisors.

Reply:
This is not true. In (Z/6)[X], 2x + 3 is not a zero divisor (because of the `mem_nonZeroDivisors_iff` lemma above), but every coefficient of it is a zero divisor.

Reply:
Yeah, sure. Sorry for the noise.